worlds/rogue_legacy/__init__.py

```diff
@@ -33,35 +33,32 @@ class RLWorld(World):
     But that's OK, because no one is perfect, and you don't have to be to succeed.
     """
     game = "Rogue Legacy"
-    option_definitions = rl_options
+    options_dataclass = RLOptions
+    options: RLOptions
     topology_present = True
     required_client_version = (0, 3, 5)
     web = RLWeb()
 
-    item_name_to_id = {name: data.code for name, data in item_table.items()}
-    location_name_to_id = {name: data.code for name, data in location_table.items()}
-
-    # TODO: Replace calls to this function with "options-dict", once that PR is completed and merged.
-    def get_setting(self, name: str):
-        return getattr(self.multiworld, name)[self.player]
+    item_name_to_id = {name: data.code for name, data in item_table.items() if data.code is not None}
+    location_name_to_id = {name: data.code for name, data in location_table.items() if data.code is not None}
 
     def fill_slot_data(self) -> dict:
-        return {option_name: self.get_setting(option_name).value for option_name in rl_options}
+        return self.options.as_dict(*[name for name in self.options_dataclass.type_hints.keys()])
```

Comment:
Just figured it was worth mentioning that this adds progression balancing, accessibility, etc. into slot_data. It also changes the way the additional names are put into slot data, changing them from sets to lists

Reply:
Testing some more, it seems like the additional name option doesn't matter because it doesn't work in the first place :(

Reply:
I'm pretty sure Phar intends to deprecate the additional name options in favor of a local override, so it may have already had that applied.